UnityProject/Assets/Scripts/ScriptableObjects/RP/EmoteBehaviors/InhaleOnEmote.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Collections.Generic;
+using HealthV2;
+using Items.Implants.Organs;
+using Systems.Atmospherics;
+using UnityEngine;
+
+namespace ScriptableObjects.RP.EmoteBehaviors
+{
+	public class InhaleOnEmote : IEmoteBehavior
+	{
+		public float Efficiency = 1.25f;
+
+		public void Behave(GameObject actor)
+		{
+			if (actor == null || actor.TryGetComponent<LivingHealthMasterBase>(out var health) == false) return;
+			var gas = GasMix.GetEnvironmentalGasMixForObject(actor.GetUniversalObjectPhysics());
+			var lungs = GetLungs(health);
+			if (lungs == null || gas == null) return;
+			foreach (var lung in lungs)
+			{
+				lung.BreatheIn(gas, health.reagentPoolSystem.BloodPool, Efficiency);
```

Comment:
You need to pull out the calculation for how much blood it's processing from the lunga, since this bypasses it

Reply:
wdym?

Reply:
ReagentMix availableBlood =
                ReagentCirculatedComponent.AssociatedSystem.BloodPool.Take(
                    (ReagentCirculatedComponent.AssociatedSystem.BloodPool.Total * efficiency) / 2f);

Reply:
that's inside of the lungs

Reply:
this wont work because both the inhale and exhale use entirely different values.

Reply:
nah tthey share the same pool

Reply:
it literally doesn't work, i just tested it